components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -0,0 +1,380 @@
+		}
+
+		if (!this.childView) {
+			this.addEventListener('focus', this.__focusCapture, true);
+			this.addEventListener('focusout', this.__focusOutCapture, true);
+			window.addEventListener('resize', this.__onWindowResize);
+		}
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		requestAnimationFrame(() => {
+			this.__autoSize(this);
+
+			if (this.getActiveView() === this) {
+				this.__fireViewResize();
+			}
+		});
+	}
+
+	getActiveView() {
+		const rootView = this.getRootView();
+		const childViews = rootView.querySelectorAll('[child-view][shown]');
+		if (!childViews || childViews.length === 0) {
+			return rootView;
+		}
+		for (let i = 0; i < childViews.length; i++) {
+			const childView = childViews[i];
+			if (childView.isActive()) {
+				return childView;
+			}
+		}
+		return rootView;
+	}
+
+	getRootView() {
+		if (!this.childView) {
+			return this;
+		}
+		const rootView = findComposedAncestor(
+			this.parentNode,
+			(node) => {
+				return node.isHierarchicalView && !node.childView;
+			}
+		);
+		return rootView;
+	}
+
+	hide(data, sourceView) {
+		if (!sourceView) {
+			sourceView = this;
+		}
+		const eventDetails = {
+			bubbles: true,
+			composed: true,
+			detail: {
+				data: data,
+				isSource: sourceView === this,
+				sourceView: sourceView
+			}
+		};
+		this.dispatchEvent(new CustomEvent('d2l-hierarchical-view-hide-start', eventDetails));
+	}
+
+	/**
+	 * Determines whether the view is active.
+	 * @return {Boolean}
+	 */
+	isActive() {
+		const content = this.shadowRoot.querySelector('.d2l-hierarchical-view-content');
+		if (this.childView && !this.shown) {
+			return false;
+		} else {
+			return !content.classList.contains('d2l-child-view-show');
+		}
+	}
+
+	/**
+	 * Forces resize/layout of the view.
+	 * @return {Boolean}
+	 */
+	resize() {
+		this.__fireViewResize();
+	}
+
+	/**
+	 * Shows the view (hiding its parent view). To show, simply call show(); on the view.
+	 */
+	show(data, sourceView) {
+		const _show = (data, view) => {
+			view.shown = true;
+			const eventDetails = {
+				bubbles: true,
+				composed: true,
+				detail: {
+					isSource: sourceView === this,
+					data: data,
+					sourceView: sourceView
+				}
+			};
+			view.dispatchEvent(new CustomEvent('d2l-hierarchical-view-show-start', eventDetails));
+		};
+
+		const _hideChildViews = (data, view) => {
+			const childViews = view.querySelectorAll('[child-view][shown]');
+			for (let i = 0; i < childViews.length; i++) {
+				childViews[i].hide(data);
+			}
+			this.resize();
+		};
+
+		if (sourceView) {
+			_show(data, this);
+			return;
+		}
+
+		sourceView = this;
+		const activeView = this.getActiveView();
+
+		if (isComposedAncestor(activeView, this)) {
+			_show(data, this);
+			return;
+		}
+
+		if (isComposedAncestor(this, activeView)) {
+			_hideChildViews(data, this);
+			return;
+		}
+
+		_hideChildViews(data, this.getRootView());
+		_show(data, this);
+
+	}
+
+	__autoSize(view) {
+		if (this.childView) {
```

Comment:
Nit: it's up to you... I think we've started to relax our practice of always using braces for conditional statements.  If you prefer, I think it's fine to go...
```
if (this.childView) return;
if (view.offsetParent === null) return;
```

Reply:
Yeah I've started using the no-brace syntax for exactly this type of scenarios -- `if (something) return;`. Anything more complex though, I personally prefer double-braces.